### PR TITLE
feat(platform/gitea): configurable autodiscover repo sorting

### PR DESCRIPTION
**Pull request by viceice**
## Changes
Allow configure the autodiscover server-side sorting for gitea.

## Context
This will allow recently updated repos faster. Not updated repos will probably skipped anyways.
https://try.gitea.io/api/swagger#/repository/repoSearch

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
@rarkins thinking of making it configurable via experimental env option?

File: docs/usage/self-hosted-experimental.md [hunk 1]
```diff
@@ -63,6 +63,28 @@ Source: [AWS s3 documentation - Interface BucketEndpointInputConfig](https://doc
 
 If set, Renovate will terminate the whole process group of a terminated child process spawned by Renovate.
 
+## `RENOVATE_X_AUTODISCOVER_REPO_SORT`
+
+<!-- prettier-ignore -->
+!!! note
+    Gitea only
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-experimental.md:70]:
```suggestion
    For the Gitea platform only.
```

File: docs/usage/self-hosted-experimental.md [hunk 2]
```diff
@@ -63,6 +63,28 @@ Source: [AWS s3 documentation - Interface BucketEndpointInputConfig](https://doc
...
+
+The sort method for autodiscover server side repository search.
+
+Allowed values: `alpha`, `created`, `updated`, `size`, `id`
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-experimental.md:74]:
```suggestion
Allowed values:

- `alpha`
- `created`
- `updated`
- `size`
- `id`
```

File: docs/usage/self-hosted-experimental.md [hunk 3]
```diff
@@ -63,6 +63,28 @@ Source: [AWS s3 documentation - Interface BucketEndpointInputConfig](https://doc
...
+Default value: `alpha`
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-experimental.md:75]:
```suggestion
Default value: `alpha`.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
Some layout changes.

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**